src/components/UpdateSubscription/components/FreeExtension.tsx

```diff
@@ -0,0 +1,199 @@
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useAppDispatch, useAppSelector } from 'redux/store';
+import { hidePopup } from 'redux/popupSlice';
+import {
+  fetchApplyRetentionAction,
+  selectRetentionActions
+} from 'redux/retentionActionsSlice';
+import Button from 'components/Button';
+import Loader from 'components/Loader';
+import {
+  ContentStyled,
+  TitleStyled,
+  TextStyled,
+  ButtonWrapperStyled
+} from 'components/InnerPopupWrapper/InnerPopupWrapperStyled';
+import { LinkStyled } from 'components/ThankYouPage/ThankYouPageStyled';
+import {
+  ImageStyled,
+  ImageWrapper
+} from 'components/SwitchPlanPopup/SwitchPlanPopupStyled';
+import { ReactComponent as Close } from 'assets/images/errors/close.svg';
+import checkmarkIcon from 'assets/images/checkmarkBase';
+
+import {
+  FreeExtensionWrapperStyled,
+  TextWrapperStyled,
+  FreeExtensionCardStyled,
+  AcceptButtonWrapperStyled,
+  FreeExtensionCardPeriodStyled
+} from './FreeExtension.styled';
+
+type FreeExtensionProps = {
```

Comment:
move to separate file

Reply:
Since it's only one type, maybe I will do it like that for now:
```
const FreeExtension = ({
  handleUnsubscribe
}: {
  handleUnsubscribe: () => void;
})
```